File: packages/canvas/src-native/canvas-android/canvas/src/main/java/com/github/triniwiz/canvas/SVGView.java
package com.github.triniwiz.canvas;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.util.AttributeSet;
import android.util.Log;
import android.view.View;
import androidx.annotation.Nullable;

/**
 * Created by triniwiz on 2019-08-05
 */
public class SVGView extends View {
    private static native long nativeInit(long svgCanvas, Bitmap bitmap);

    private static native long drawSVG(long svgCanvas, Bitmap bitmap, String svg);

    public Bitmap bitmap;
    public long svgCanvas = 0;
    boolean isInit;

    public SVGView(Context context) {
        super(context);
    }

    public SVGView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onSizeChanged(int w, int h, int oldw, int oldh) {
        if (!isInit) {
            bitmap = Bitmap.createBitmap(w, h, Bitmap.Config.ARGB_8888);
            svgCanvas = SVGView.nativeInit(svgCanvas, bitmap);
            isInit = true;
        } else {
            // scale (w / oldw) (h / oldh)
        }
    }

    @Override
    protected void onDraw(Canvas canvas) {
        canvas.drawBitmap(bitmap, 0, 0, null);
    }

    public void setSrc(String src) {
        svgCanvas = SVGView.drawSVG(svgCanvas, bitmap, src);
        invalidate();
    }
}

File: packages/canvas/src-native/canvas-android/canvas/src/main/java/com/github/triniwiz/canvas/SVGView.kt
package com.github.triniwiz.canvas

import android.content.Context
import android.graphics.Bitmap
import android.graphics.Canvas
import android.util.AttributeSet
import android.view.View

/**
 * Created by triniwiz on 2019-08-05
 */
class SVGView : View {
	internal var bitmap: Bitmap? = null
	private var svgCanvas: Long = 0
	private var isInit = false

	constructor(context: Context?) : super(context) {}
	constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
		handler
	}

	override fun onSizeChanged(w: Int, h: Int, oldw: Int, oldh: Int) {
		if (!isInit) {
			bitmap = Bitmap.createBitmap(w, h, Bitmap.Config.ARGB_8888)
			svgCanvas = nativeInit(svgCanvas, bitmap)
			isInit = true
		} else {
			// scale (w / oldw) (h / oldh)
		}
	}

	override fun onDraw(canvas: Canvas) {
		canvas.drawBitmap(bitmap!!, 0f, 0f, null)
	}

	fun setSrc(src: String) {
		svgCanvas = drawSVG(svgCanvas, bitmap, src)
		invalidate()
	}

	companion object {
		@JvmStatic
		private external fun nativeInit(svgCanvas: Long, bitmap: Bitmap?): Long
		@JvmStatic
		private external fun drawSVG(svgCanvas: Long, bitmap: Bitmap?, svg: String): Long
	}
}
